comp/vision/gamepiece_finder24.py

```diff
@@ -0,0 +1,204 @@
+        self.object_higher = (255, 170, 255)
+        self.scale_factor = 1
+        self.width = camera_params[0]
+        self.height = camera_params[1]
+        self.object_height = .105
+        self.theta = 0
+        self.topic_name = topic_name
+        self.initialize_nt()    
+        self.output_stream = CameraServer.putVideo("Processed", self.width, self.height)
+
+    def initialize_nt(self):
+        """Start NetworkTables with Rio as server, set up publisher."""
+        self.inst = NetworkTableInstance.getDefault()
+        self.inst.startClient4("retro-finder")
+        # this is always the RIO IP address; set a matching static IP on your
+        # laptop if you're using this in simulation.
+        self.inst.setServer("10.1.0.2")
+        # Table for vision output information
+        topic_name = "vision/" + self.serial
+        self.vision_fps = self.inst.getDoubleTopic(topic_name + "/fps").publish()
+        self.vision_latency = self.inst.getDoubleTopic(
+            topic_name + "/latency"
+        ).publish()
+
+        # work around https://github.com/robotpy/mostrobotpy/issues/60
+        self.inst.getStructTopic("bugfix", NotePosition).publish().set(
+            NotePosition(Translation2d())
+        )
+
+        self.vision_nt_struct = self.inst.getStructArrayTopic(
+            topic_name + "/NotePosition", NotePosition
+        ).publish()
+
+    def find_object(self, img):
+        range = cv2.inRange(img, self.object_lower, self.object_higher)
+        img_rgb = cv2.cvtColor(img, cv2.COLOR_HSV2RGB)
+        floodfill = range.copy()
+        h, w = range.shape[:2]
+        mask = np.zeros((h+2, w+2), np.uint8)
+        cv2.floodFill(floodfill, mask, (0,0), 255)
+        floodfill_inv = cv2.bitwise_not(floodfill)
+        img_floodfill = range | floodfill_inv
+        median = cv2.medianBlur(img_floodfill, 5)
+        contours, hierarchy = cv2.findContours(
+            median, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
+        objects = {}
+        objects["objects"] = []
+        for c in contours:
+            _, _, cnt_width, cnt_height = cv2.boundingRect(c)
+            if (cnt_height < 50):
+                continue
+            if (cnt_height/cnt_width < 2):
+                continue
+            if (cnt_height/cnt_width > 5):
+                continue
+            mmnts = cv2.moments(c)
+            if (mmnts["m00"] == 0):
+                continue
+
+            cX = int(mmnts["m10"] / mmnts["m00"])
+            cY = int(mmnts["m01"] / mmnts["m00"])
+            
+            # translation_x = (cX-self.width/2)* \
+            #     (self.object_height*math.cos(self.theta)/cnt_height)
+            # translation_y = (cY-self.height/2) * \
+            #     (self.object_height*math.cos(self.theta)/cnt_height)
+            # translation_z = (self.object_height*self.scale_factor*math.cos(self.theta))/(cnt_height)
+
+            object = []
+
+            objects["objects"].append(
```

Comment:
what's this "objects" key for?   don't you just want to append to the objects array?